tests/unit/unit-walker.cpp

```diff
@@ -115,6 +173,490 @@ TEST_P(UnitWalkerTests, FunSubstitution)
   EXPECT_EQ(fy, iw.visit(fx));
 }
 
+void printVector(vector<int> v){
+  for (int i:v){
+    cout << v[i] << ", " << endl;
+  }
+}
+
+string vectorToString(vector<int> v){
+  string s;
+  for (int i:v){
+    s += to_string(i);
+    s += ", ";
+  }
+  return s;
+}
+
+
+void printMap(UnorderedTermPairMap const &m){
+  for (auto const& pair : m){
+    auto val_pair = pair.second;
+    cout << pair.first << ": " << "<" << val_pair.first << ", (" << vectorToString(val_pair.second) << ")>" << " }\n";
+    //cout << "{ " << pair.first << ": " << pair.second << " }" << endl;
+  }
+}
+
+TEST_P(UnitWalkerTests, SimpleTree)
```

Comment:
In each test:
1. general documentation of what each block of code does.
2. When you create a term, document what term are you creating. For example, before the block of code that creates the term `x+y=x+1`, write a comment like:
```
// construct `x+y=x+1`
```
No need to do this for the sub-terms, though. 

2. When you create an "expected" variable (vector/map/whatever), plz document how it looks like in a comment, e.g.: 
```
x+y-> (x+y, [])
x -> (x+y, [0]
...
```